addons/website_sale/models/website_snippet_filter.py

```diff
@@ -167,7 +167,8 @@ def _get_products_recently_sold_with(
         self, website, limit, domain, product_template_id, **kwargs,
     ):
         products = self.env['product.product']
-        current_template = self.env['product.template'].browse(int(product_template_id)).exists()
+        current_template = product_template_id and \
+            self.env['product.template'].browse(int(product_template_id)).exists()
```

Comment:
```suggestion
        current_template = product_template_id \
            and self.env['product.template'].browse(int(product_template_id)).exists()
```
Sorry, had it wrong myself when I first suggested it, but sale guidelines are specific about the `and` being on the start of the next line in cases like these